src/sc_kpm/sc_sets/sc_oriented_set.py

```diff
@@ -0,0 +1,88 @@
+    The other have edges between edges from set_node marked with 'nrel_basic_sequence'.
+    """
+
+    def add(self, *elements: ScAddr) -> None:
+        """Add elements to oriented-sc-set"""
+        if elements:
+            elements_iterator = iter(elements)
+            if self.is_empty():
+                current_edge = create_role_relation(self._set_node, next(elements_iterator), ScKeynodes.rrel_index(1))
+            else:
+                current_edge = self._get_last_edge()
+            for next_element in elements_iterator:
+                next_edge = create_edge(sc_types.EDGE_ACCESS_CONST_POS_PERM, self._set_node, next_element)
+                create_norole_relation(current_edge, next_edge, ScKeynodes[CommonIdentifiers.NREL_BASIC_SEQUENCE])
+                current_edge = next_edge
+
+    def _get_last_edge(self) -> Optional[ScAddr]:
+        start_template = search_role_relation_template(self._set_node, ScKeynodes.rrel_index(1))
+        next_edge = start_template.get(ScAlias.ACCESS_EDGE)
+        while True:
+            elem_search_result = self._search_next_element_template(next_edge)
+            if elem_search_result is None:
+                return next_edge
+            next_edge = elem_search_result.get(ScAlias.ACCESS_EDGE)
+
+    def _search_next_element_template(self, cur_element_edge: ScAddr) -> Optional[ScTemplateResult]:
+        templ = ScTemplate()
+        templ.triple_with_relation(
+            cur_element_edge,
+            sc_types.EDGE_D_COMMON_VAR,
+            sc_types.EDGE_ACCESS_VAR_POS_PERM >> ScAlias.ACCESS_EDGE,
+            sc_types.EDGE_ACCESS_VAR_POS_PERM,
+            ScKeynodes[CommonIdentifiers.NREL_BASIC_SEQUENCE],
+        )
+        templ.triple(self._set_node, ScAlias.ACCESS_EDGE, sc_types.UNKNOWN >> ScAlias.ELEMENT)
+        search_results = template_search(templ)
+        if not search_results:
+            return None
+        return search_results[0]
+
+    def __iter__(self) -> Iterator[ScAddr]:
+        """Iterate by oriented-sc-set elements"""
+        start_template = search_role_relation_template(self._set_node, ScKeynodes.rrel_index(1))
+        if not start_template:
+            return
+        yield start_template.get(ScAlias.ELEMENT)
+        next_edge = start_template.get(ScAlias.ACCESS_EDGE)
+        while True:
+            elem_search_result = self._search_next_element_template(next_edge)
+            if elem_search_result is None:
+                return
+            yield elem_search_result.get(ScAlias.ELEMENT)
+            next_edge = elem_search_result.get(ScAlias.ACCESS_EDGE)
+
+    @property
+    def elements_list(self) -> List[ScAddr]:
+        """List of elements with order"""
+        return list(self)
+
+    def remove(self, *elements: ScAddr) -> None:
+        """Clear and add existing elements without given ones"""
+        # TODO: optimize
+        elements_new = [element for element in self.elements_list if element not in elements]
```

Comment:
This logic repeated. Can you implement common interface for set classes?

Reply:
This logic must be unique later